src/ApiServer/ApiServer/Service/MerchantService.cs

```diff
@@ -45,7 +46,7 @@ public enum MerchantListOrder {
                 filters.Add(Builders<Merchant>.Filter.Eq($"{nameof(Merchant.Access)}.{nameof(AccessControlEntry<MerchantRole>.UserId)}", controlledBy.Value));
             }
             if(!string.IsNullOrWhiteSpace(textSearch)) {
-                filters.Add(Builders<Merchant>.Filter.Text(textSearch, new TextSearchOptions { CaseSensitive = false, DiacriticSensitive = false }));
+                filters.Add(Builders<Merchant>.Filter.Regex(m => m.Name, regex: new BsonRegularExpression(Regex.Escape(textSearch), "i")));
```

Comment:
Questo va benissimo come soluzione temporanea al problema di ricerca (ricordo che c'erano alcune stringhe di ricerca che non ritornavano dati utili, nonostante fosse ragionevole aspettarsi dei match). Idealmente però la ricerca testuale dovrebbe essere più efficiente della ricerca tramite regex in MongoDB: proviamo a vedere più avanti se riusciamo a trovare una soluzione migliore 🤔